### PR TITLE
Feature#71/sidebar design

### PR DESCRIPTION
**Pull request by jairolan**
# Description
- I have added the sidebar layout for all screens.
- I have added a js file for button interaction.
- I have removed a css line to apply a general styling structure later.
- I have added the authenticate_user helper in the product controller.

### CODE REVIEW

**Review by oramasangel — APPROVED** [review]
lgtm

**Review by JoseRicardoJimenez — DISMISSED** [review]
OK!

**Review by xFerchoVJ — APPROVED** [review]

File: local_sales/app/javascript/packs/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const btn = document.querySelector(".mobile-menu-button");
+const sidebar = document.querySelector(".sidebar");
+
+btn.addEventListener("click", () => {
+  sidebar.classList.toggle("-translate-x-full");
+});
```
Comment (YilKanda) [inline local_sales/app/javascript/packs/sidebar.js:6]: Add a line blank.

File: local_sales/app/views/layouts/_sidebar.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  <!-- mobile menu bar -->
+  <% if user_signed_in? %>
+    <div class="bg-gray-800 text-gray-100 flex justify-between md:hidden">
+      <!-- logo -->
+      <a href="#" class="block p-4 text-white font-bold">Local Sales</a>
+      <!-- mobile menu-button -->
+      <button class="mobile-menu-button p-4 focus:outline-none focus:bg-gray-700">
+        <svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 6h16M4 12h16M4 18h16" />
+        </svg>
```
Comment (YilKanda) [inline local_sales/app/views/layouts/_sidebar.html.erb:11]: Can you call this SVG externally?

File: local_sales/app/views/layouts/_sidebar.html.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      </button>
+    </div>
+    <!-- sidebar -->
+    <div class="sidebar bg-blue-800 text-blue-100 w-64 space-y-6 py-7 px-2 absolute inset-y-0
+      left-0 transform -translate-x-full md:relative md:translate-x-0 transition duration-200 ease-in-out">
+      <!-- logo -->
+      <a href="#" class="text-white flex items-center space-x-2 px-4">
+        <svg class="h-8 w-8" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 12l2 2 4-4M7.835 4.697a3.42 3.42 0 001.946-.806 3.42 3.42 0 014.438 0 3.42 3.42 0 001.946.806 3.42 3.42 0 013.138 3.138 3.42 3.42 0 00.806 1.946 3.42 3.42 0 010 4.438 3.42 3.42 0 00-.806 1.946 3.42 3.42 0 01-3.138 3.138 3.42 3.42 0 00-1.946.806 3.42 3.42 0 01-4.438 0 3.42 3.42 0 00-1.946-.806 3.42 3.42 0 01-3.138-3.138 3.42 3.42 0 00-.806-1.946 3.42 3.42 0 010-4.438 3.42 3.42 0 00.806-1.946 3.42 3.42 0 013.138-3.138z" />
+        </svg>
```
Comment (YilKanda) [inline local_sales/app/views/layouts/_sidebar.html.erb:21]: Can you call this SVG externally?

**Review by YilKanda — COMMENTED** [review]
Leave small comments.

**Review by xFerchoVJ — APPROVED** [review]

**Review by juliocabrera820 — APPROVED** [review]

**Review by oramasangel — APPROVED** [review]
lgtm

**Pull request merged**